// !WITH_NEW_INFERENCE

class A {
    fun bar() {
        <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
            <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                    <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                        <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                            <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                    <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                        <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                            <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                    <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                        <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                            <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                                <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                                    <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                                        <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                                            <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                                                <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                                                    <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                                                        <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                                                            <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                                                                <!NI;NONE_APPLICABLE!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!>div<!><!> {
                                                                                                    <!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>+<!><!>(<!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>foo<!><!> ?: "")
                                                                                                    <!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>+<!><!>(<!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>foo<!><!> ?: "")
                                                                                                    <!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>+<!><!>(<!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>foo<!><!> ?: "")
                                                                                                    <!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>+<!><!>(<!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>foo<!><!> ?: "")
                                                                                                    <!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>+<!><!>(<!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>foo<!><!> ?: "")
                                                                                                    <!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>+<!><!>(<!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>foo<!><!> ?: "")
                                                                                                    <!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>+<!><!>(<!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>foo<!><!> ?: "")
                                                                                                    <!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>+<!><!>(<!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>foo<!><!> ?: "")
                                                                                                    <!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>+<!><!>(<!NI;UNRESOLVED_REFERENCE!><!UNRESOLVED_REFERENCE!>foo<!><!> ?: "")
                                                                                                }
                                                                                            }
                                                                                        }
                                                                                    }
                                                                                }
                                                                            }
                                                                        }
                                                                    }
                                                                }
                                                            }
                                                        }

                                                    }
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
